src/components/CheckEditor/CheckEditor.tsx

```diff
@@ -119,6 +120,18 @@ export default class CheckEditor extends PureComponent<Props, State> {
     this.setState({ check });
   };
 
+  onProbeOptionsChange = ({ timeout, frequency, probes }: OnChangeArgs) => {
+    const { check } = this.state;
+    this.setState({
+      check: {
+        ...check,
+        timeout,
+        frequency,
+        probes,
+      },
+    });
+  };
```

Comment:
Using the current state value like this can be dangerous because it may not be up to date. And so whenever using the current state value, it should be done through an [updater function](https://reactjs.org/docs/react-component.html#setstate).
(Also e.g. `onSettingsUpdate` above. I'll leave it up to you to see whether there are even more cases. ;) )

Reply:
Good call!